pkg/operator/controller.go

```diff
@@ -1045,97 +1042,20 @@ func (c *Controller) run(stopCh <-chan struct{}) {
 	}
 
 	klog.V(1).Info("starting events processor")
-	go wait.Until(c.eventProcessor, time.Second, stopCh)
+	go wait.Until(c.eventProcessor, time.Second, ctx.Done())
 	klog.Info("started events processor/controller")
 
-	<-stopCh
+	<-ctx.Done()
 	c.enableNodeInformer(false)
 	c.enablePodInformer(false)
 	klog.Info("shutting down events processor/controller")
 }
 
-// BecomeLeader uses client-go leader-with-lease election mechanism to
-// select a leader and starts the controller by calling operator run
-// method.
-func (c *Controller) BecomeLeader(stopCh <-chan struct{}) {
-	// Become the leader before proceeding.
-	klog.Info("trying to become a leader")
-
-	// Check the presence of the legacy node-tuning-operator-lock ConfigMap.
-	// If the owner references exist in the ConfigMap, we have an old garbage
-	// collection based locking scheme and we cannot continue until the ConfigMap
-	// is GC or deleted manually.  If the owner references do not exist, just go
-	// ahead with the new client-go leader-election code.
-loop:
```

Comment:
@jmencak   In case the move to the controller runtime wrapper does not handle this GC, will we still need to keep this cleanup for legacy NTO lock ?  
What is the upgrade path for NTO ?  can we skip major versions and miss this cleanup ?

Reply:
So this cleanup code was added in 4.8.  In my opinion we no longer need to keep it since any of the 4.8, 4.9, 4.10 versions will take care of this cleanup.  In other words, we don's support direct upgrades from say 4.7->4.11.